calculator_test.go

```diff
@@ -2,23 +2,202 @@ package calculator_test
+		{inputs: []float64{100}, want: 100, name: "Difference of single argument"},
+		{inputs: []float64{1, 1, 1}, want: -1, name: "Difference of 3 positive numbers to give a negative number"},
+		{inputs: []float64{5, 1, 1}, want: 3, name: "Difference of 3 positive numbers to give a positive number"},
+		{inputs: []float64{-1, -1, -1, -1}, want: 2, name: "Difference of 4 negative numbers to give a positive number"},
+	}
+
+	for _, tc := range testCases {
+		got := calculator.Subtract(tc.inputs...)
+		if tc.want != got {
+			t.Errorf("%s: Subtract(%v) want %f, got %f", tc.name, tc.inputs, tc.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+	rand.Seed(time.Now().UnixNano())
+	numTestCases := 100
+	maxInputValue := 10
+	maxNumInputsPerTestCase := 10
+	testCases := make(chan *variadicTestCase)
+
+	go func() {
+		for i := 0; i < numTestCases; i++ {
+			numInputs := rand.Intn(maxNumInputsPerTestCase)
+			inputs := []float64{}
+			var want float64
+
+			if numInputs > 0 {
+				want = 1
+			}
+
+			for j := 0; j < numInputs; j++ {
+				randomInput := float64(rand.Intn(maxInputValue) + 1)
+				want *= randomInput
+				inputs = append(inputs, randomInput)
+			}
+			testCases <- &variadicTestCase{inputs: inputs, want: want}
+		}
+		close(testCases)
+	}()
+
+	for tc := range testCases {
+		got := calculator.Multiply(tc.inputs...)
+		if tc.want != got {
+			t.Errorf("Multiply(%v) want %f, got %f", tc.inputs, tc.want, got)
+		}
+	}
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	testCases := []*variadicErrorTestCase{
+		{inputs: []float64{}, want: 0, errExpected: false, name: "Division of empty set of inputs to return an error"},
+		{inputs: []float64{2}, want: 2, errExpected: false, name: "Division with only 1 input to return an error"},
+		{inputs: []float64{4, 2}, want: 2, errExpected: false, name: "Division of evenly divisible positive number to give a positive quotient"},
+		{inputs: []float64{-4, -2}, want: 2, errExpected: false, name: "Division of 2 negative numbers (evenly divisible) to give a positive quotient"},
+		{inputs: []float64{-4, 2, 1}, want: -2, errExpected: false, name: "Division of negative number by even number to give negative quotient"},
+		{inputs: []float64{4, 0, 1}, want: 0, errExpected: true, name: "Division by zero to return an error"},
+		{inputs: []float64{0, 1}, want: 0, errExpected: false, name: "Dividing 0 by another non-zero number to give 0 as the quotient"},
+		{inputs: []float64{2, 4}, want: 0.5, errExpected: false, name: "Dividing a smaller number by larger number to give a decimal quotient"},
+	}
+
+	for _, tc := range testCases {
+		got, err := calculator.Divide(tc.inputs...)
+
+		if tc.errExpected && err == nil {
+			t.Fatalf("%s: Divide(%v) expected an error to be returned but got nil", tc.name, tc.inputs)
+		}
+
+		if !tc.errExpected && err != nil {
+			t.Fatalf("%s: Divide(%v) returned an unexpected error: %v", tc.name, tc.inputs, err)
+		}
+
+		if !tc.errExpected && tc.want != got {
```

Comment:
Good! We mustn't even _compare_ `want` and `got` unless this is a "valid" test case (that is, unless no error is expected). Well done.